public/app/features/explore/Wrapper.tsx

```diff
@@ -38,6 +39,20 @@ class WrapperUnconnected extends PureComponent<Props> {
     lastSavedUrl.left = undefined;
     lastSavedUrl.right = undefined;
 
+    // timeSrv (which is used internally) on init reads `from` and `to` param from the URL and updates itself
+    // using those value regardless of what is passed to the init method.
```

Comment:
> ...using those value regardless of what is passed to the init method.

Is it true though? There's a swap on `init` https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/services/TimeSrv.ts#L73

Reply:
from `init`:
```ts
    this.dashboard = dashboard;
    this.time = dashboard.time;
    this.refresh = dashboard.refresh;

    this.initTimeFromUrl();
    this.parseTime();
```

(`dashboard` is the only parameter `init` gets). first `this.time` is populated from `dashboard.time`, then in `this.initTimeFromUrl` it gets replaced by whatever is in the URL.

and since when it swaps variables it also updates the URL this means if you happen to call `init` with "inverted" `from` and `to` you get locked to the first "corrected" time range

Reply:
to make it short, we need to:

- make sure there are no `from` and `to` in the URL, so that whatever we pass to init is used in timeSrv
- make sure `from` and `to` are always chronologically ordered to avoid timeSrv to perform the swapping and trigger https://github.com/grafana/grafana/blob/c23bc1e7b79d01a1c13d3e66ffa88ebad3ef6be6/public/app/features/dashboard/services/TimeSrv.ts#L303


to be clear, i don't really like the solution and i think that if we can't make `timeSrv` side-effect free we should probably implement something that suits Explore needs better.

Reply:
Coolio, makes sense, thanks for the explanation 👍